flow-typed/fusion-core.js

```diff
@@ -51,16 +51,17 @@ declare module 'fusion-core' {
     renderer: any;
     cleanup(): Promise<any>;
     enhance<Token, Deps>(token: Token, enhancer: Function): void;
-    register: any;
-    // register<Deps, Provides>(Plugin: FusionPlugin<Deps, Provides>): aliaser<*>;
-    // register<Token, Deps>(
-    //   token: Token,
-    //   Plugin: FusionPlugin<Deps, $Call<ExtractReturnType, Token>>
-    // ): aliaser<*>;
-    // register<Token: Object>(
-    //   token: Token,
-    //   val: $Call<ExtractReturnType, Token>
-    // ): aliaser<*>;
+    register<Deps, Provides>(Plugin: FusionPlugin<Deps, Provides>): aliaser<*>;
+    register<Token, Deps>(
+      token: Token,
+      Plugin: FusionPlugin<Deps, $Call<ExtractReturnType, Token>>
+    ): aliaser<*>;
+    register<Token: Object>(
+      token: Token,
+      val: $Call<ExtractReturnType, Token>
+    ): aliaser<*>;
+    // Register a plugin /w middleware and no dependencies
+    register(val: ?() => void): aliaser<*>;
```

Comment:
I didn't think we allowed this. Are you sure this last bit is needed?

Reply:
I believe this fixed an error before, but I can't remember where. We can try removing it.